subprojects/core/src/main/java/org/gradle/internal/fingerprint/impl/AbstractFileCollectionFingerprinter.java

```diff
@@ -44,7 +48,11 @@ public AbstractFileCollectionFingerprinter(FingerprintingStrategy fingerprinting
     @Override
     public CurrentFileCollectionFingerprint fingerprint(FileCollection files) {
         List<FileSystemSnapshot> roots = fileCollectionSnapshotter.snapshot(files);
-        return DefaultCurrentFileCollectionFingerprint.from(roots, fingerprintingStrategy);
+        CurrentFileCollectionFingerprint fingerprint = DefaultCurrentFileCollectionFingerprint.from(roots, fingerprintingStrategy);
+        if (logger.isDebugEnabled()) {
```

Comment:
I think the logging should go here: https://github.com/gradle/gradle/blob/9131ce509dfa9ea78615374aab3802593bdfd384/subprojects/execution/src/main/java/org/gradle/internal/execution/caching/impl/LoggingCachingStateBuilder.java#L58

So it is emitted when enabling the build cache debug logging. WDYT?

Reply:
I was using this for debugging task classpath inputs so I didn't want to limit it to the build cache, but could certainly add logging there too?

Reply:
The logging is not limited to inputs of cacheable tasks. It will be emitted whenever the build cache is enabled for all tasks and all inputs. So I think it would give you what you want.

The difference is that the build cache logging can be enabled separately from the debug logging. And what you are adding here seems to only make sense if you try to debug build cache misses, right?

Reply:
What I mean is - the fingerprinter is used for `@Classpath` inputs so if the logging is limited to the build cache you'd need to enable the build cache to debug a normalization problem with a task classpath input.

Reply:
I don't fully understand. What is a normalization problem with the task classpath input which is not build-cache related? If the task is not up-to-date on a build with no changes, then `--info` will give you detailed information about the file which changed.

If it is build cache related, then why not run with `--build-cache`. That is all that is necessary.

Reply:
I thought that task up-to-date checking used the same fingerprinting and normalization as the build cache. Is that not the case?